lib/new_relic/agent/instrumentation/roda/roda_transaction_namer.rb

```diff
@@ -0,0 +1,39 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      module Roda
+        module TransactionNamer
+          extend self
+
+          def initial_transaction_name(request)
+            transaction_name(::NewRelic::Agent::UNKNOWN_METRIC, request)
+          end
+
+          ROOT = '/'.freeze
+          REGEX_MUTIPLE_SLASHES = %r{^[/^\\A]*(.*?)[/\$\?\\z]*$}.freeze
+
+          def transaction_name(path, request)
+            verb = http_verb(request)
+            path = request.path if request.path
+            name = path.gsub(REGEX_MUTIPLE_SLASHES, '\1') # remove any rogue slashes
```

Comment:
what does the `'\1'` mean? Does it just replace things with literally a `\1` in the string? I'm just curious about what this is for since it's not ringing any bells for me.

Reply:
The `\1` is referencing the first capture group or the stuff inside the first set of parentheses. If we had two capture groups, then we'd get to that second capture group with `\2`

So we're saying to look for any amount of beginning and ending `\`, capture only the text between, and replace the whole thing with the first capture group.